### PR TITLE
chore: update deps after v0.19.0 release

### PR DESCRIPTION
**Pull request by galargh**
This PR was created to conclude the v0.19.0 release - https://github.com/ipfs/kubo/issues/9502. We should aim to land it ahead of the 1.20 release - https://github.com/ipfs/kubo/issues/9694.

@Jorropo adding you as reviewer for visibility because I wouldn't want to make the boxo consolidation/rename efforts harder for you with this.

### CODE REVIEW

**Comment by guseggert** [comment]
> I wouldn't want to make the boxo consolidation/rename efforts harder for you with this.

Thankfully this doesn't have any meaningful impact on that, so LGTM.

**Review by guseggert — APPROVED** [review]

**Comment by guseggert** [comment]
I've also covered https://github.com/ipfs/kubo/pull/9744/commits/3cf234d0c725edd33a22c46a7bd915a23188e7f7 in https://github.com/ipfs/kubo/pull/9736, but it's fine to leave here too

**Review by hacdias — APPROVED** [review]
LGTM, updated with `master`. Will merge if ✅

**Pull request merged**
